controller/handler_ctrl/connect.go

```diff
@@ -98,16 +59,54 @@ func (self *ConnectHandler) HandleConnection(hello *channel2.Hello, certificates
 		KeyUsages:     []x509.ExtKeyUsage{x509.ExtKeyUsageAny},
 	}
 
+	hasValidCert := false
 	var errorList errorz.MultipleErrors
 
 	for _, cert := range certificates {
 		if _, err := cert.Verify(opts); err == nil {
-			return nil
+			hasValidCert = true
 		} else {
 			errorList = append(errorList, err)
 		}
 	}
 
-	//goland:noinspection GoNilness
-	return errorList.ToError()
+	if !hasValidCert && len(errorList) > 0 {
+		return errorList.ToError()
+	}
+
+	log := pfxlog.Logger().WithField("routerId", hello.IdToken)
+
+	fingerprint := ""
+	log.Debugf("peer has [%d] certificates", len(certificates))
+	for i, c := range certificates {
```

Comment:
we should filter out certificates that failed the check (`cert.Verify()`)above. 
in general this logic is suspect if multiple certs are present, since we only take one fingerprint

Reply:
I'm thinking we could check against all valid certs. That way a client could present old and new certs

Reply:
change made